### PR TITLE
add attention mask; fix stop_str length

### PR DESCRIPTION
**Pull request by xzzWZY**

### CODE REVIEW

File: scripts/benchmark.py [hunk 1]
```diff
@@ -206,21 +218,25 @@ def run_inference(
 
         # deal with stop_str
         if stop_str:
+            stop_str_length = np.zeros(batch_size, dtype=int)
             if isinstance(stop_str, str):
                 pos_array = np.char.rfind(output_np, stop_str, rfind_start)
                 find_stop = pos_array != -1
+                stop_str_length[find_stop] = len(tokenizer(stop_str).input_ids)
```
Comment (jaywonchung) [inline scripts/benchmark.py]: Compute the length of `stop_str` outside the loop. It's shared by every iteration and every prompt.

File: pegasus/benchmark.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # {{ gpu }} is defined in `hosts.yaml`, and will be filled in when Pegasus
 # determines the specific node and gpu the generated job command will run on.
 - command:
-    - docker exec leaderboard{{ gpu }} python scripts/benchmark.py --input-file sharegpt/sg_90k_part1_html_cleaned_lang_first_sampled_sorted.json --model-path {{ model }} --task {{ task }}
+    - docker exec leaderboard{{ gpu }} python scripts/benchmark.py --input-file sharegpt/sg_90k_part1_html_cleaned_lang_first_sampled_sorted.json --model-path {{ model }} --task {{ task }} --batch-size {{ batch }}
```
Comment (jaywonchung) [inline pegasus/benchmark.yaml]:
```suggestion
    - docker exec leaderboard{{ gpu }} python scripts/benchmark.py --input-file sharegpt/sg_90k_part1_html_cleaned_lang_first_sampled_sorted.json --model-path {{ model }} --task {{ task }} --batch-size {{ batch_size }}
```

File: pegasus/benchmark.yaml [hunk 3]
```diff
@@ -31,3 +31,9 @@
     - chat-concise
     - instruct
     - instruct-concise
+  batch:
```
Comment (jaywonchung) [inline pegasus/benchmark.yaml]:
```suggestion
  batch_size:
```

**Review by jaywonchung — CHANGES_REQUESTED** [review]
Thanks for you work. Two questions and one performance-related change request.

1. What happened to the sharegpt dataset?
2. Just to confirm -- the decoder of an encoder-decoder model does not take attention_mask?

**Comment by xzzWZY** [comment]
1. previously sharegpt is sorted but the shortest sequence at front, now longest at front
2. if I understand right, the default value of decoder_attention_mask for encoder-decoder model is used

_decoder_attention_mask (torch.BoolTensor of shape (batch_size, target_sequence_length), optional) — Default behavior: generate a tensor that ignores pad tokens in decoder_input_ids. Causal mask will also be used by default._

from https://huggingface.co/docs/transformers/v4.33.0/en/model_doc/encoder-decoder#transformers.EncoderDecoderModel.forward.decoder_attention_mask

**Comment by jaywonchung** [comment]
Could you mention that in sharegpt/README.md? Also I think somehow sort.py is missing. It would be somewhere in your local directory.

**Review by jaywonchung — APPROVED** [review]
Looks good! Thanks for the quick update. But please monitor the benchmark JSON files as they run.
